src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -767,58 +831,203 @@ public IEnumerable<string> GetCollections(Transaction transaction)
+                        }
+                        else
+                        {
+                            throw new VoronErrorException($"Failed to remove revision with change vector {revisionInfo.ChangeVector}");
+                        }
+                    }
 
-                        hasValue = true;
+                    Table writeTable = null;
+                    if (writeTables.TryGetValue(revisionInfo.CollectionName, out writeTable) == false)
+                    {
+                        writeTable = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, new CollectionName(revisionInfo.CollectionName));
+                        writeTables[revisionInfo.CollectionName] = writeTable;
+                    }
+                    writeTable.DeleteByKey(keySlice);
+                }
+            }
 
-                        using (Slice.From(context.Allocator, revision.ChangeVector, out var keySlice))
-                        {
-                            CreateTombstone(context, keySlice, revision.Etag, collectionName, changeVector, lastModifiedTicks);
+            return maxEtagDeleted;
+        }
 
-                            maxEtagDeleted = Math.Max(maxEtagDeleted, revision.Etag);
-                            if ((revision.Flags & DocumentFlags.HasAttachments) == DocumentFlags.HasAttachments)
-                            {
-                                _documentsStorage.AttachmentsStorage.DeleteRevisionAttachments(context, revision, changeVector, lastModifiedTicks);
-                            }
+        private long HandleConflictRevisionsDelete(DocumentsOperationContext context, Table table, Slice prefixSlice,
+            CollectionName collectionName,
+            string changeVector, long revisionCount, long lastModifiedTicks, bool handleNotConflicted,
+            bool skipForceCreated,
+            out bool moreWork)
+        {
+            var revisionsToRemove = new List<(string ChangeVector, long Etag, DocumentFlags Flags, string CollectionName)>();
+
+            var config = ConflictConfiguration.Default;
+
+            var conflictCount = GetConflictRevisionsCount(context, table, prefixSlice);
+            var regularCount = revisionCount - conflictCount;
+            var regularDeletedCount = 0;
+            var conflictDeletedCount = 0;
+
+            long skippedForceCreated = 0;
+
+            var reachedNotOutOfDate = false;
+            moreWork = false;
+            var finishedRegualr = false;
+            var finishedConflicted = false;
+
+            foreach (var read in table.SeekForwardFrom(RevisionsSchema.Indexes[IdAndEtagSlice], prefixSlice, skip: 0, startsWith: true))
+            {
+                if (config.MaximumRevisionsToDeleteUponDocumentUpdate.HasValue &&
+                    config.MaximumRevisionsToDeleteUponDocumentUpdate.Value <= regularDeletedCount + conflictDeletedCount)
```

Comment:
so in that `ConflictedRevisionsDeletionState` you can have a method called something like
`ReachedMaximumRevisionsToDeleteUponDocumentUpdate`

Reply:
Done